CHANGELOG.md

```diff
@@ -250,128 +80,100 @@
+- Make various protected `Node` methods public [[#2544]](https://github.com/aiidateam/aiida-core/pull/2544)
+- Renaming `DbNode.type` to `DbNode.node_type` [[#2552]](https://github.com/aiidateam/aiida-core/pull/2552)
+- Rename the ORM classes for `Node` sub classes `JobCalculation`, `WorkCalculation`, `InlineCalculation` and `FunctionCalculation` [[#2184]](https://github.com/aiidateam/aiida-core/pull/2184)[[#2189]](https://github.com/aiidateam/aiida-core/pull/2189)[[#2192]](https://github.com/aiidateam/aiida-core/pull/2192)[[#2195]](https://github.com/aiidateam/aiida-core/pull/2195)[[#2201]](https://github.com/aiidateam/aiida-core/pull/2201)
+- Do not allow the `copy` or `deepcopy` of `Node`, except for `Data`  [[#1705]](https://github.com/aiidateam/aiida-core/pull/1705)
+- Reorganization of some top level modules [[#2357]](https://github.com/aiidateam/aiida-core/pull/2357)
+- Make the node repository API backend agnostic [[#2506]](https://github.com/aiidateam/aiida-core/pull/2506)
+- Redesign the Parser class [[#2397]](https://github.com/aiidateam/aiida-core/pull/2397)
+- [Django]: Support for datetime objects removed from node attributes and extras [[#3090]](https://github.com/aiidateam/aiida-core/pull/3090)
+- Enforce specific precision in `clean_value` for floats when computing a node's hash [[#3108]](https://github.com/aiidateam/aiida-core/pull/3108)
+- Remove physical constants from `aiida.common.constants` to `qe-tools` package [[#3278]](https://github.com/aiidateam/aiida-core/pull/3278)
+- Add type checks to all plugin factories [[#3456]](https://github.com/aiidateam/aiida-core/pull/3456)
+- Disallow pickle when storing numpy array in `ArrayData` [[#3434]](https://github.com/aiidateam/aiida-core/pull/3434)
+- Remove implementation of legacy workflows [[#2379]](https://github.com/aiidateam/aiida-core/pull/2379)
+- Implement `CalcJob` process class that replaces the deprecated `JobCalculation` [[#2389]](https://github.com/aiidateam/aiida-core/pull/2389)
+- Redefine the structure of the `CalcInfo.local_copy_list` [[#2581]](https://github.com/aiidateam/aiida-core/pull/2581)
+- Change 'ancestor_of'/'descendant_of' to 'with_descendants'/'with_ancestors' [[#2278]](https://github.com/aiidateam/aiida-core/pull/2278)
 
-### Developers
-- Move requirements from setup_requirements.py to setup.json [[#2307]](https://github.com/aiidateam/aiida-core/pull/2307)
-- Refactor the management of the configuration file [[#2349]](https://github.com/aiidateam/aiida-core/pull/2349)
-- Add a transaction context manager to backend [[#2387]](https://github.com/aiidateam/aiida-core/pull/2387)
-- Enable tab-completion for `verdi devel tests` [[#1809]](https://github.com/aiidateam/aiida-core/pull/1809)
-- Add `-v/--verbose` flag to `verdi devel tests` [[#1807]](https://github.com/aiidateam/aiida-core/pull/1807)
 
 
-## v0.12.4
```

Comment:
That wasn't so much the reason I did it, it was more because for the subsub sections in the "Detailed list of changes" would have 4 hashes. I am not sure how that would show. If that still has sub headings that are fine, I am fine to revert these changes. Is there a way I can preview the Github rendering?

Reply:
I'm sure it'll work fine